app/models/timed_service_price_policy.rb

```diff
@@ -0,0 +1,31 @@
+class TimedServicePricePolicy < PricePolicy
+
+  include PricePolicies::Usage
+
+  CHARGE_FOR = { usage: "usage" }.freeze
+
+  def estimate_cost_and_subsidy_from_order_detail(order_detail)
+    return if order_detail.quantity.blank?
+
+    estimate_cost_and_subsidy(order_detail.quantity)
+  end
+
+  def calculate_cost_and_subsidy_from_order_detail(order_detail)
+    calculate_for_time(order_detail.quantity)
+  end
+
+  def estimate_cost_and_subsidy(duration)
+    return if restrict_purchase?
+
+    calculate_for_time(duration)
+  end
+
+  private
+
+  def calculate_for_time(duration)
+    start_at = Time.current.beginning_of_day
+    end_at = start_at + duration.minutes
+    PricePolicies::TimeBasedPriceCalculator.new(self).calculate(start_at, end_at)
```

Comment:
Something about this feels awkward to me. We're making times that don't really have a meaning. I would probably lean toward doing a straight `{ cost: usage_rate * quantity, usage_subsidy * quantity }`. That also avoids an awkward check for `schedule_rules` in the calculator.